ambry-router/src/main/java/com.github.ambry.router/PutManager.java

```diff
@@ -259,24 +290,28 @@ void completePendingOperations() {
    * by the {@link ReadableStreamChannel} associated with the operation.
    */
   private class ChunkFiller implements Runnable {
-    private final int sleepTimeWhenIdleMs = 10;
-
     public void run() {
       try {
         while (isOpen.get()) {
-          boolean allChunksFilled = true;
+          boolean idleIteration = true;
           for (PutOperation op : putOperations) {
             if (!op.isChunkFillComplete()) {
               op.fillChunks();
-              allChunksFilled = false;
+              idleIteration = false;
```

Comment:
if any work is done in an iteration, we do not want to attempt the sleep as there is a high likelihood that work could get done in the next iteration as well.